kvwrapper_etcd/kvwrapper_etcd.go

```diff
@@ -52,7 +67,11 @@ func (e EtcdWrapper) GetVal(key string) (*kvwrapper.KeyValue, error) {
 
 // GetVal returns a []KeyValue found at key
 func (e EtcdWrapper) GetList(key string, sort bool) ([]*kvwrapper.KeyValue, error) {
-	r, err := e.client.Get(key, sort, true)
+	options := &etcd.GetOptions{
+		Sort:      sort,
+		Recursive: true,
+	}
+	r, err := e.kapi.Get(context.Background(), key, options)
 	if err != nil {
 		if strings.HasPrefix(err.Error(), "501:") {
```

Comment:
Yeah I think we're doing a bit too much interpretation of the errors. I think what we need to distinguish is between when we connect to etcd and the key is simply not found, which is a pretty normal occurrence and failing to connect to etcd or get a value from it for any one of a multitude of reasons.

Unfortunately we have some FD code that is checking specifically for `	if err == kvwrapper.ErrCouldNotConnect {`, but I think that's a bad idea and I'm going to send a PR shortly to fix that. Once have that, then I think it makes sense to simply bubble the error up without interpretation, which I think is better because we don't throw away specific info about the error, like we're doing now.